common/types/epoch.go

```diff
@@ -11,49 +11,53 @@ import (
 // EpochID is the running epoch number. It's zero-based, so the genesis epoch has EpochID == 0.
 type EpochID uint32
 
+func (e EpochID) Uint32() uint32 {
+	return uint32(e)
+}
+
 // EncodeScale implements scale codec interface.
-func (l EpochID) EncodeScale(e *scale.Encoder) (int, error) {
-	n, err := scale.EncodeCompact32(e, uint32(l))
+func (e EpochID) EncodeScale(enc *scale.Encoder) (int, error) {
+	n, err := scale.EncodeCompact32(enc, uint32(e))
 	if err != nil {
 		return 0, err
 	}
 	return n, nil
 }
 
 // DecodeScale implements scale codec interface.
-func (l *EpochID) DecodeScale(d *scale.Decoder) (int, error) {
-	value, n, err := scale.DecodeCompact32(d)
+func (e *EpochID) DecodeScale(dec *scale.Decoder) (int, error) {
+	value, n, err := scale.DecodeCompact32(dec)
 	if err != nil {
 		return 0, err
 	}
```

Comment:
Again unrelated, but this is definitely wrong. To produce the error some bytes of the underlying `io.Reader`/`scale.Decoder` must have been consumed.